client/src/commands/editConnectionCommand.ts

```diff
@@ -13,128 +13,228 @@
 */
 'use strict';
 import * as vscode from 'vscode';
+import * as fs from 'fs-extra';
 import { UserInputUtil, IBlockchainQuickPickItem } from './UserInputUtil';
 import { FabricConnectionRegistryEntry } from '../fabric/FabricConnectionRegistryEntry';
 import { FabricConnectionRegistry } from '../fabric/FabricConnectionRegistry';
+import { ConnectionPropertyTreeItem } from '../explorer/model/ConnectionPropertyTreeItem';
 import { ConnectionTreeItem } from '../explorer/model/ConnectionTreeItem';
 import { FabricConnectionHelper } from '../fabric/FabricConnectionHelper';
-import { ConnectionPropertyTreeItem } from '../explorer/model/ConnectionPropertyTreeItem';
+import { IFabricWalletGenerator } from '../fabric/IFabricWalletGenerator';
+import { FabricWalletGeneratorFactory } from '../fabric/FabricWalletGeneratorFactory';
+import { ExtensionUtil } from '../util/ExtensionUtil';
 import { ParsedCertificate } from '../fabric/ParsedCertificate';
 import { VSCodeOutputAdapter } from '../logging/VSCodeOutputAdapter';
 import { LogType } from '../logging/OutputAdapter';
+import { IFabricWallet } from '../fabric/IFabricWallet';
 
 export async function editConnectionCommand(treeItem: ConnectionPropertyTreeItem | ConnectionTreeItem): Promise < {} | void > {
+
```

Comment:
there are alot of if's and else's in here maybe make smaller functions so its easier to see whats going on?